test/sharness/t0100-name.sh

```diff
@@ -60,6 +60,15 @@ test_expect_success "ipfs cat on published content succeeds" '
     test_cmp expected actual
 '
 
+# test with a custom key
+test_expect_success "ipfs resolves published key from keystore" '
```

Comment:
Lets break this test out into a few more steps.

First assertion should just generate a key (and not have the || true bit, if it fails we want to fail).

Second assertion should do the publish, and check its output

Third assertion should do the resolve